### PR TITLE
Use micromamba 2.0.5.rc0 in Dockerimage

### PR DESCRIPTION
**Pull request by JohanMabille**
This is just for testing the latest pre-release of micromamba (i.e. 2.0.5.rc0) before releasing for good. I don't know if this is enough, let me know if additional changes are required.

### CODE REVIEW

**Comment by mathbunnyru** [comment]
> I don't know if this is enough, let me know if additional changes are required.

This is enough 👍

**Comment by mathbunnyru** [comment]
@JohanMabille you probably want to change the version here as well:
https://github.com/jupyter/docker-stacks/blob/main/images/docker-stacks-foundation/Dockerfile#L128

**Comment by JohanMabille** [comment]
@mathbunnyru ah indeed, thanks a lot!

File: images/docker-stacks-foundation/Dockerfile [hunk 1]
```diff
@@ -115,17 +115,19 @@ RUN set -x && \
...
     ./bin/micromamba install \
+        -c conda-forge \
+        -c conda-forge/label/mamba_prerelease \
         --root-prefix="${CONDA_DIR}" \
         --prefix="${CONDA_DIR}" \
         --yes \
         'jupyter_core' \
         'conda' \
-        'mamba' \
+        'mamba=2.0.5.rc0' \
```
Comment (jjerphan) [inline images/docker-stacks-foundation/Dockerfile:130]: For the paranoid:
```suggestion
        'conda-forge/label/mamba_prerelease::mamba=2.0.5.rc0' \
```

Comment (JohanMabille) [reply images/docker-stacks-foundation/Dockerfile:130]: The logs show that mamba 2.0.5.rc0 is installed

**Comment by JohanMabille** [comment]
@mathbunnyru For a reason I don't understand, it seems that the failing tests are using mamba 1 instead of mamba 2. This looks  like a caching issue in the CI.

**Comment by JohanMabille** [comment]
Ok let's close this and open a new PR.

**Pull request closed without merging**